packages/headless/src/features/sort-criteria/criteria.ts

```diff
@@ -150,44 +143,84 @@ export const criterionDefinition = new RecordValue({
   },
 });
 
-/**
- * Utility function that validates and builds a valid criterion. Throws an error when the criterion is invalid.
- * @param criterion The criterion definition.
- * @returns A criterion.
- */
-export function validateSortCriterion(criterion: {
-  by: SortBy;
+function parseCriterion(criterion: {
+  by: string;
   order?: SortOrder;
-  field?: string;
 }): SortCriterion {
-  new Schema({criterion: criterionDefinition}).validate({criterion});
-  const {by, field, order} = criterion;
+  const {by, order} = criterion;
 
   switch (by) {
     case SortBy.Relevancy:
       return buildRelevanceSortCriterion();
     case SortBy.QRE:
       return buildQueryRankingExpressionSortCriterion();
+    case SortBy.NoSort:
+      return buildNoSortCriterion();
     case SortBy.Date:
       if (!order) {
         throw new Error(
           '"order" should be specified when "by" value is "date"'
         );
       }
       return buildDateSortCriterion(order);
-    case SortBy.Field:
-      if (!field) {
-        throw new Error(
-          '"field" should be specified when "by" value is "field"'
-        );
-      }
+    default:
       if (!order) {
         throw new Error(
-          '"order" should be specified when "by" value is "field"'
+          `"order" should be specified when "by" value is "field" for field "${by}"`
         );
       }
-      return buildFieldSortCriterion(field, order);
-    default:
-      return buildNoSortCriterion();
+      return buildFieldSortCriterion(by, order);
   }
 }
+
+function isSortOrder(order?: string): order is SortOrder {
+  return (
+    order === undefined ||
+    order === SortOrder.Ascending ||
+    order === SortOrder.Descending
+  );
+}
+
+/**
+ * Parses a criterion expression and return a list of `SortCriterion`
+ * @param expression Sort criterion expression
+ *
+ * The available sort criteria are:
+ * - `relevancy`
+ * - `date ascending`/`date descending`
+ * - `qre`
+ * - `field ascending`/`field descending`, where you must replace `field` with the name of a sortable field in your index (e.g., `criteria="size ascending"`).
+ *
+ * You can specify multiple sort criteria to be used in the same request by separating them with a comma (e.g., `criteria="size ascending, date ascending"` ).
+ */
+export function parseCriterionExpression(expression: string) {
```

Comment:
This whole function could (and should) be super easily unit testable.

Also, it should be "tolerant", as much as possible of whitespace when it can.

eg:
```javascript
"size ascending, date ascending"
"size ascending,date ascending"
" size ascending, date ascending "
```

if possible, should produce the same result.

Reply:
yep, it's coming in a different PR

Reply:
I would prefer to place the parser in it's own file e.g. `criteria-parser.ts` to help separate concerns, and then have the unit tests sit along-side `criteria-parser.test.ts`.